packages/charts/src/hooks/useSetMarginLeft.ts

```diff
@@ -0,0 +1,14 @@
+export const useSetMarginLeft = (dataset, labelKey) =>
```

Comment:
As said in the above comment, I would rather name this `useChartMargin` and return a full, memoized margin object here. - [ ] use `useMemo`, dependent on `dataset` and `labelKey`
- [ ] Could you split this one-call-chain into multiple calls? I think it is very hard to read.
- [ ] Instead of multiplying the length with 4, you could also make use of the `getTextWidth` method from the Utils. This will get you the exact length